components/server/src/bitbucket-server/bitbucket-server-api.ts

```diff
@@ -190,6 +191,50 @@ export class BitbucketServerApi {
         );
     }
 
+    async getBranchLatestCommit(
+        user: User,
+        params: {
+            repoKind: "projects" | "users" | string;
+            owner: string;
+            repositorySlug: string;
+            branch: string;
+        },
+    ): Promise<BitbucketServer.Branch | undefined> {
+        // @see https://developer.atlassian.com/server/bitbucket/rest/v811/api-group-repository/#api-api-latest-projects-projectkey-repos-repositoryslug-branches-get
+        // @see https://bitbucket.gitpod-dev.com/rest/api/1.0/users/huiwen/repos/mustard/branches?filterText=develop
+        const queryParam = qs.stringify({
+            filterText: params.branch,
```

Comment:
Not sure if it's `true` by default or not, but we could include `boostMatches` to both improve performance and fix edge cases with having many branches which kindof match pushing the exact match to a further page in the pagination

Reply:
https://developer.atlassian.com/server/bitbucket/rest/v811/api-group-repository/#api-api-latest-projects-projectkey-repos-repositoryslug-branches-get-request-Query%20parameters

Reply:
Addressed 🙏